force-app/test/ADDR_Addresses_TEST.cls

```diff
@@ -2111,6 +2110,188 @@ public with sharing class ADDR_Addresses_TEST {
+
+        Test.startTest();
+        insert testContact;
+        Test.stopTest();
+
+        Address__c testAddress = [SELECT Latest_Start_Date__c, Default_Address__c
+                                  FROM Address__c
+                                  WHERE MailingState__c = 'Washington'
+                                  LIMIT 1];
+        System.assertEquals(true, testAddress.Default_Address__c, 'Address should be set as the default address.');
+        System.assertEquals(System.today(), testAddress.Latest_Start_Date__c, 'The default address latest start date ' +
+                'should be set to today\'s date.');
+    }
+
+    @IsTest
+    private static void latestStartDateOnUpdatedDefaultAddressSetToToday() {
+        if (ADV_PackageInfo_SVC.useAdv()) {
+            return;
+        }
+        buildTestAddress(2);
+        NPSP_Contact npspContact = new NPSP_Contact(UTIL_UnitTestData_TEST.getContact());
+        npspContact.mailingAddress(new NPSP_Address(testAddresses[0]));
+
+        insert npspContact.getRecord();
+
+        Contact testContact = [SELECT Current_Address__c, Current_Address__r.Default_Address__c, AccountId
+                               FROM Contact
+                               WHERE Id = :npspContact.getRecord().Id
+                               LIMIT 1];
+        Id defaultAddressId = testContact.Current_Address__c;
+        System.assertEquals(true, testContact.Current_Address__r.Default_Address__c, 'The address should be the default address of ' +
+                'the household.');
+
+        testAddresses[1].Household_Account__c = testContact.AccountId;
+        insert testAddresses[1];
+
+        Address__c nonDefaultAddress = [SELECT Default_Address__c FROM Address__c WHERE Id = :testAddresses[1].Id];
+
+        Test.startTest();
+            nonDefaultAddress.Default_Address__c = true;
+            update nonDefaultAddress;
+        Test.stopTest();
+
+        Address__c newDefaultAddress = [SELECT Default_Address__c, Latest_Start_Date__c, Latest_End_Date__c
+                                        FROM Address__c
+                                        WHERE Id = :nonDefaultAddress.Id
+                                        LIMIT 1];
+        Address__c previousDefaultAddress = [SELECT Default_Address__c, Latest_Start_Date__c, Latest_End_Date__c
+                                             FROM Address__c
+                                             WHERE Id = :defaultAddressId
+                                             LIMIT 1];
+        System.assertEquals(false, previousDefaultAddress.Default_Address__c, 'The previous default address should be' +
+                ' set to false.');
+        System.assertEquals(true, newDefaultAddress.Default_Address__c, 'The new default address should be' +
+                ' set to true.');
+        System.assertEquals(System.today(), previousDefaultAddress.Latest_End_Date__c, 'The latest end date on the ' +
```

Comment:
I'm getting picky, but can we also assert the Latest Start Date for the former default? It should not be changed in this case.